aiida/restapi/resources.py

```diff
@@ -96,6 +96,7 @@ class BaseResource(Resource):
 
     _translator_class = BaseTranslator
     _parse_pk_uuid = None  # Flag to tell the path parser whether to expect a pk or a uuid pattern
+    _method_decorators = [close_session]  # Close SQLA session after any method call
```

Comment:
That is a nifty feature of `flask` and a nice solution. Since we want this decorator to apply to all resource methods and all subclasses, why not just define it straight as class attribute `method_decorators = [close_session]` here instead of applying this private attribute in the constructor? This way there is also no problems with subclasses forgetting to call `super` if they override the constructor

Reply:
It is to keep the current API (sort of) still there. It seemed one could pass the keyword argument `get_decorators` when adding a resource to the REST API (using `.add_resource()` in `aiida.restapi.api:AiidaApi`). To keep this functionality, and potentially add more decorators from there, I have renamed the keyword argument to simply `decorators`, since it doesn't conform to simply to HTTP `GET` request any more.
I wanted to align the two ways of passing decorators to the HTTP request methods, but unfortunately that's not possible. Either `method_decorators` is a list with callables, and they apply to _all_ HTTP request methods OR `method_decorators` is a dictionary/mapping, where the keys are the HTTP request method names and the values a list of callable decorators.

Reply:
Hmm.. re-reading your comment here, it seems my answer is based on a misunderstanding. And yeah sure.
However, it also means that if a sub-class "forgets" to call `super().__init__()` it will not get the benefit of the `decorators` keyword argument.